hibernate-core/src/test/java/org/hibernate/jpa/test/procedure/StoreProcedureOutParameterByNameTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.jpa.test.procedure;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Id;
+import javax.persistence.NamedStoredProcedureQuery;
+import javax.persistence.ParameterMode;
+import javax.persistence.StoredProcedureParameter;
+import javax.persistence.StoredProcedureQuery;
+import javax.persistence.Table;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.hibernate.dialect.Oracle10gDialect;
+import org.hibernate.engine.jdbc.connections.spi.JdbcConnectionAccess;
+import org.hibernate.engine.jdbc.spi.JdbcServices;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.jpa.test.BaseEntityManagerFunctionalTestCase;
+import org.hibernate.testing.FailureExpected;
+import org.hibernate.testing.RequiresDialect;
+import org.hibernate.testing.TestForIssue;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * @author Andrea Boriero
+ * @author Gail Badner
+ */
+@TestForIssue(jiraKey = "HHH-10756")
+@RequiresDialect(Oracle10gDialect.class)
+public class StoreProcedureOutParameterByNameTest extends BaseEntityManagerFunctionalTestCase {
+	EntityManagerFactory entityManagerFactory;
+
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class[] {User.class};
+	}
+
+	@Before
+	public void startUp() {
+		entityManagerFactory = getOrCreateEntityManager().getEntityManagerFactory();
+
+		createProcedures( entityManagerFactory );
+	}
+
```

Comment:
The following test fails for Oracle because the input/output parameters are named, but Hibernate binds the input parameter by name and binds the output parameter by position. Oracle does not allow "mixed" parameters.